subtests/docker_cli/login/login.py

```diff
@@ -161,6 +161,9 @@ def docker_login(self, password=None):
         """
         if password is None:
             password = self.sub_stuff['passwd']
+        # FIXME: --email option is deprecated in 1.12, and will be
+        # removed in 1.13. We still need it for <= 1.10; otherwise
+        # docker prompts from stdin, and we hang.
```

Comment:
> We really, really need to figure out a way to log all output and capture unexpected warnings

``DockercmdBase`` would be the place.  Maybe a method that fires off a thread that dumps ``docker log -f`` output into ``logging.debug()`` - autotest will scoop that up automatically.  Another option would be to dump it into a file in ``self.subtest.resultsdir`` (I think that's the attribute name).